### ISSUE TITLE
loss=0.44 on the SROIE dataset

### ISSUE THREAD

**Issue by ghost** [courao/ocr.pytorch#6]
@courao 
I have been training for 1 day on the [SROIE dataset.](https://rrc.cvc.uab.es/?ch=13), the loss is still 0.44 ! 
It works well on other datasets, but not the SROIE dataset?
Am I doing something wrong?
[dataset download link](https://github.com/dhiraa/text-information-extraction#dataset)

![X00016469620](https://user-images.githubusercontent.com/49877135/64128638-27ffc800-cdc0-11e9-9bac-f36a21859996.jpg)
![X00016469622](https://user-images.githubusercontent.com/49877135/64128653-3817a780-cdc0-11e9-9de9-460bfd8f33d8.jpg)

**Comment by courao** [comment]
Maybe the size of the image seems quite thin?
You can try to adjust the input size of images for training (such as making the height larger).
And during inference, you can also make the height larger.

**Comment by ghost** [comment]
here is a [sample.zip](https://github.com/opconty/pytorch_ctpn/files/3570491/sample.zip)
or you can download the [entire dataset](https://drive.google.com/drive/folders/1ShItNWXyiY1tFDM5W02bceHuJjyeeJl2)

What do you think is causing this issue?

**Comment by courao** [comment]
I changed the height value to 1440 in ctpn_predict.py, and find the results seem not bad by the pretrained model. You can have a try or change it to a larger value!
![X00016469612](https://user-images.githubusercontent.com/12403083/64244810-48ed2800-cf3c-11e9-939a-e8353f04ccf8.jpg)
![X00016469619](https://user-images.githubusercontent.com/12403083/64244812-48ed2800-cf3c-11e9-951b-49c205bd0f65.jpg)
![X00016469620](https://user-images.githubusercontent.com/12403083/64244813-4985be80-cf3c-11e9-9e5f-49334cfa52bb.jpg)
![X00016469622](https://user-images.githubusercontent.com/12403083/64244814-4985be80-cf3c-11e9-885d-024dd5585b5f.jpg)
![X00016469623](https://user-images.githubusercontent.com/12403083/64244815-4985be80-cf3c-11e9-9b91-0ec83d192801.jpg)
![X00016469669](https://user-images.githubusercontent.com/12403083/64244816-4a1e5500-cf3c-11e9-902f-70e21b93c372.jpg)
![X00016469672](https://user-images.githubusercontent.com/12403083/64244817-4a1e5500-cf3c-11e9-9515-73a823d27d63.jpg)

**Comment by ghost** [comment]
@courao hmmmmm....
After I changed values to `height = 1440` in `ctpn_predict.py`, the results were almost perfect for the trained data, and it even was able to detect some images that it didn't see before. But, the new issue is that some images that it was able detect at `height = 720` perfectly, now at `height = 1440` it can't.

So my main question is:
is there a way to make ctpn detect multi-height & multi-scale images, without having to set it ourselves.

Notice that some images better detected at 1440, and some are better detected at 720:
Left side: `height = 1440`
Right side: `height = 720`

![1](https://user-images.githubusercontent.com/49877135/64266822-ef85f880-cf3d-11e9-937b-6197528ec38e.jpg)
![2](https://user-images.githubusercontent.com/49877135/64266824-ef85f880-cf3d-11e9-93a9-e2e93fc42eb6.jpg)
![3](https://user-images.githubusercontent.com/49877135/64266825-ef85f880-cf3d-11e9-941a-2c619a8d2a26.jpg)
`

**Comment by ghost** [comment]
@opconty courao seems to be on it.
The main issue seems to be that the heights/widths  in `ctpn_predict.py` are fixed for all images.
The solution would be Multi-scale detection, allowing each image to have a different height/width.

**Comment by opconty** [comment]
> @opconty courao seems to be on it.
> The main issue seems to be that the heights/widths in `ctpn_predict.py` are fixed for all images.
> The solution would be Multi-scale detection, allowing each image to have a different height/width.

ok,  I'll try

**Comment by courao** [comment]
@deepseek Main reason for the different behaviors with various heights is that, during training, I set the maximum value of the long side of the image to 1600 in case of cuda memory error.
Therefore I recommend that the long side should not be too large (no more than 1600) or too small while testing, otherwise the too large or too small text area will be missed.

Another way to automatically fit the text area is using multi-scale test with nms, but required more computation load.

**Comment by ghost** [comment]
> Another way to automatically fit the text area **is using multi-scale test with nms, but required more computation load.**

How to use multi-scale test with nms?

**Comment by ghost** [comment]
@courao This issue is very important.
Because different height gets different results.
Can you integrate a solution that solves this issue

**Comment by courao** [comment]
@deepseek the way to use multi-scale test with nms is that firstly using images of different sizes as input and getting predicted anchors with confidence probabilities, then rescale the images to a uniform scale(the anchor size needs to be rescaled as well), finally apply nms by the confidence.
This may increase the time for detection and I'm not sure if this idea will work. And I'm so sorry that I've been working on another study recently, if you have time you can try this.
